### PR TITLE
Set default number of columns for `terminal` and `code` cheat sheets to 2

### PR DESCRIPTION
**Pull request by sahildua2305**
###### What does your Pull Request do (check all that apply)?

Choose the most relevant items and use the following title template to name
your Pull Request.
- [ ] New Instant Answer
  - [ ] Cheat Sheets: **`New {Cheat Sheet Name} Cheat Sheet`**
  - [ ] Other: **`New {IA Name} Instant Answer`**
- [ ] Improvement
  - [ ] Bug fix: **`{IA Name}: Fix {Issue number or one-line description}`**
  - [ ] Enhancement: **`{IA Name}: {Description of Improvements}`**
- [x] Non–Instant Answer
  - [x] Other (Role, Template, Test, Documentation, etc.): **`{GoodieRole/Templates/Tests/Docs}: {Short Description}`**
###### Description of changes

Change default number of columns for `template` and `code` template cheat sheets to 2 (two).
###### Testing

Check layout of the following cheat sheets once this change is live on Beta.
- [Htaccess Cheat Sheet](https://beta.duckduckgo.com/?q=htaccess+cheat+sheet&t=hd&ia=cheatsheet&iax=1) (`Terminal`)
- [Git-SSH Cheat Sheet](https://beta.duckduckgo.com/?q=git+ssh+cheat+sheet&t=hd&ia=cheatsheet&iax=1) (`Terminal`)
- [GitHub Markdown Cheat Sheet](https://beta.duckduckgo.com/?q=gfm+cheat+sheet&t=hd&ia=cheatsheet&iax=1) (`Code`)
###### People to notify (@mention interested parties)

@moollaza (#3353)

https://duck.co/ia/view/cheat_sheets

### CODE REVIEW

**Comment by daxtheduck** [comment]
Metadata placeholder post
            
*If you haven't already, please edit your first comment to include a link to the IA page at the bottom.*
            
---
            
**How to find your IA page**
- New Instant Answer
  -- follow the steps [here](https://duck.co/ia/new_ia) to create your IA page.
              
- Existing Instant Answer
  -- find your IA page from this [list](https://duck.co/ia)

**Comment by edgesince84** [comment]
thanks Sahil, should be a quick review!

**Comment by moollaza** [comment]
@sahildua2305 this is looking great.

One thing: It looks like GFM should be Terminal? https://github.com/duckduckgo/zeroclickinfo-goodies/blob/master/share/goodie/cheat_sheets/json/github-flavoured-markdown.json#L13

Was it changed to code? because it needs the full width of the table to display properly..

File: share/goodie/cheat_sheets/cheat_sheets.js [hunk 1]
```diff
@@ -12,7 +12,14 @@ DDH.cheat_sheets.build = function(ops) {
...
-        // Change number of columns to show, if mentioned in the cheat sheet
+        // Default columns to 2, for `template` and `code` template types
+        if (template_type == 'terminal' || template_type == 'code') {
+            showColumns = 2;
+        }
+
+        // Change number of columns to show, if mentioned in the cheat sheet.
+        // Keep this check below the template_type check (valid for `terminal` and `code`),
+        // so that default 2-column layout can be overwritten using `columns` parameter explicitly
         if (columns && columns >= 1 && columns <= 4)
```
Comment (moollaza) [inline share/goodie/cheat_sheets/cheat_sheets.js]: Should be `> 1 && columns < 4` -- we shouldn't allow 1 or 4 column layouts

**Comment by sahildua2305** [comment]
@moollaza I didn't change any template_type in any cheat sheet.

**Comment by moollaza** [comment]
@sahildua2305 this is looking good, but I just noticed that for some (e.g. https://beta.duckduckgo.com/?q=ssh+cheat+sheet&t=hd&ia=cheatsheet) we're now stacking the first two sections, which has doubled the initial height of the cheat sheet display. Unfortunately that's too tall, so we either need to force the initial columns to the standard width, or maybe only show 1 column initially, though that means we'd be removing some info from the initial display.

**Comment by sahildua2305** [comment]
@moollaza I think we can force `height` of `.compressed` to make sure it looks good in one column as well. I will have to try something up.

**Comment by mintsoft** [comment]
I'm going to close this for now (doing some house-keeping). Feel free to re-open if I've closed it in error

**Pull request closed without merging**